release.py

```diff
@@ -0,0 +1,694 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Release automation script."""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import re
+import subprocess
+from datetime import datetime
+from pathlib import Path
+
+import github
+import github.GitRelease
+import github.Repository
+
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger('release')
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+if 'GITHUB_TOKEN' not in os.environ:
+    raise SystemExit('Environment variable GITHUB_TOKEN not set.')
+
+auth = github.Auth.Token(os.environ['GITHUB_TOKEN'])
+gh_client = github.Github(auth=auth)
+
+
+VERSION_REGEX = r'(\d+\.\d+\.\d+(?:\.dev\d+)?)'
+VERSION_FILES = {
+    'ops/src': Path('ops/version.py'),
+    'ops/pyproject': Path('pyproject.toml'),
+    'testing': Path('testing/pyproject.toml'),
+    'tracing': Path('tracing/pyproject.toml'),
+    'uvlock': Path('uv.lock'),
+}
+
+
+def get_latest_release_tag(repo: github.Repository.Repository, branch_name: str) -> str | None:
+    """Get the latest release tag from the repository."""
+    releases = repo.get_releases()
+
+    maintenance_branch_version = (
+        branch_name.removesuffix('-maintenance') if branch_name.endswith('-maintenance') else None
+    )
+
+    for release in releases:
+        if release.draft:
+            continue
+
+        if maintenance_branch_version and maintenance_branch_version not in release.tag_name:
+            continue
+
+        return release.tag_name
+
+    return None
+
+
+def bump_minor_version(version: str) -> str:
+    """Bump minor version."""
+    major, minor, patch = map(int, version.split('.'))
+    return f'{major}.{minor + 1}.{patch}'
```

Comment:
Would we want to reset the patch version to 0 here? It would seem odd to bump from say 2.23.4 to 2.24.4.